console/client/src/layout/IDELayout.tsx

```diff
@@ -1,30 +1,68 @@
 import { XMarkIcon } from '@heroicons/react/24/outline'
-import { useContext } from 'react'
+import React from 'react'
 import { ModuleDetails } from '../features/modules/ModuleDetails'
 import { ModulesList } from '../features/modules/ModulesList'
 import { Timeline } from '../features/timeline/Timeline'
 import { VerbTab } from '../features/verbs/VerbTab'
-import { SelectedModuleContext } from '../providers/selected-module-provider'
-import { TabType, TabsContext } from '../providers/tabs-provider'
+import { TabsContext, TabSearchParams, TabType } from '../providers/tabs-provider'
 import { headerColor, headerTextColor, panelColor } from '../utils/style.utils'
 import { SidePanel } from './SidePanel'
+import { RequestModal } from '../features/requests/RequestsModal'
+import { useSearchParams, useNavigate , useLocation } from 'react-router-dom'
+import { Tab } from '@headlessui/react'
 
 const selectedTabStyle = `${headerTextColor} ${headerColor}`
 const unselectedTabStyle = `text-gray-300 bg-slate-100 dark:bg-slate-600`
 
 export function IDELayout() {
-  const { selectedModule } = useContext(SelectedModuleContext)
-  const { tabs,activeTab, setActiveTab, setTabs } = useContext(TabsContext)
-
+  const { tabs,activeTab, setActiveTab, setTabs } = React.useContext(TabsContext)
+  const navigate = useNavigate()
+  const location = useLocation()
+  const [ searchParams ] = useSearchParams()
+  
+ 
   const handleCloseTab = id => {
     if (activeTab === id && tabs.length > 1) {
       // Set the next available tab as active, if the current active tab is being closed
-      const nextTab = tabs.find(tab => tab.id !== id) || tabs[0]
-      setActiveTab(nextTab)
+      const index = tabs.findIndex(tab => tab.id === id)
+      setActiveTab(index - 1 )
+      searchParams.delete(TabSearchParams.active)
     }
     setTabs(tabs.filter(tab => tab.id !== id))
+    navigate({ ...location, search: searchParams.toString() })
+  }
+
+  const handleChangeTab = (index: number) => {
+    setActiveTab(index)
+    index > 0 && tabs.length > 1
+      ? searchParams.set(TabSearchParams.active, tabs[index].id)
+      : searchParams.delete(TabSearchParams.active)
+    navigate({ ...location, search: searchParams.toString() })
   }
 
+  // Handle opening the correct tab on load
+  React.useEffect(() => {
+    const id = searchParams.get(TabSearchParams.active)
+    if(!id) {
+      setActiveTab(0)
+      return
+    }
+    const index = tabs.findIndex(tab => tab.id === id)
+    if(index >= 0) {
+      setActiveTab(activeTab  ?? index)
```

Comment:
Seems like there are double spaces here `setActiveTab(activeTab  ?? index)` should be `setActiveTab(activeTab ?? index)`

Should the linter find and fix these?